static/gsApp/hooks/useMetricDetectorLimit.tsx

```diff
@@ -0,0 +1,45 @@
+import useOrganization from 'sentry/utils/useOrganization';
+import {useDetectorsQuery} from 'sentry/views/detectors/hooks/index';
+
+import useSubscription from 'getsentry/hooks/useSubscription';
+
+interface MetricDetectorLimitResponse {
+  detectorCount: number;
+  detectorLimit: number;
+  hasReachedLimit: boolean;
+  isError: boolean;
+  isLoading: boolean;
+}
+
+const UNLIMITED_QUOTA = -1;
+
+export function useMetricDetectorLimit(): MetricDetectorLimitResponse {
+  const organization = useOrganization();
+  const subscription = useSubscription();
+  const {data: detectors, isLoading, isError} = useDetectorsQuery();
```

Comment:
```ts
const {..., getResponseHeader} = useDetectorsQuery({
  query: 'type:metric'
  limit: 0
}, {
  enabled: !organization.features.includes('workflow-engine-metric-detector-limit') ||
       detectorLimit === UNLIMITED_QUOTA
})

const hits = getResponseHeader()['X-Hits'];
const totalMetricDetectors = hits ? parseInt(hits): 0;
```

This will get the max number through the header so we don't return every detector in the response data, and limit it to metric detectors. Also skips the request if it's not necessary